### PR TITLE
[FLINK-31032] Supports AND push down for orc format

### PR DESCRIPTION
**Pull request by FangYongs**
This PR aims to supports `AND` push down in orc format.

### CODE REVIEW

File: flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/filter/OrcFilters.java [hunk 1]
```diff
@@ -344,4 +344,27 @@ public String toString() {
             return "OR(" + Arrays.toString(preds) + ")";
         }
     }
+
+    /** An AND predicate that can be evaluated by the OrcInputFormat. */
+    public static class And extends Predicate {
+        private final Predicate[] preds;
+
+        public And(Predicate... predicates) {
```
Comment (SteNicholas) [inline flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/filter/OrcFilters.java:352]: Adds the java doc of the constructor which style follows the `OR`, `NOT` etc:
```
/**
   * Creates an AND predicate.
   *
   * @param predicates The disjunctive predicates.
   */
```

Comment (FangYongs) [reply flink-table-store-format/src/main/java/org/apache/flink/table/store/format/orc/filter/OrcFilters.java:357]: Thanks @SteNicholas Done

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @JingsongLi

**Comment by FangYongs** [comment]
The test fails due to a known issue https://issues.apache.org/jira/browse/FLINK-31039

**Comment by SteNicholas** [comment]
@FangYongs, please reopen the pull request and rerun CI automatically.

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**